File: src/main/java/al132/alchemistry/recipes/LiquifierRecipe.java
package al132.alchemistry.recipes;

import net.minecraft.fluid.Fluid;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fluids.FluidStack;


public class LiquifierRecipe {

    public ItemStack input;
    public FluidStack output;

    public LiquifierRecipe(ItemStack input, FluidStack output) {
        this.input = input;
        this.output = output;
    }

    public LiquifierRecipe(ItemStack input, Fluid fluid, int fluidQuantity) {
        this(input, new FluidStack(fluid, fluidQuantity));
    }
}

File: src/main/java/al132/alchemistry/recipes/LiquifierRecipe.kt
package al132.alchemistry.recipes

import net.minecraft.item.ItemStack
import net.minecraftforge.fluids.Fluid
import net.minecraftforge.fluids.FluidStack

/**
 * Created by al132 on 4/29/2017.
 */

data class LiquifierRecipe(val input: ItemStack, val output: FluidStack) {

    constructor(input: ItemStack, fluid: Fluid, fluidQuantity: Int) : this(input, FluidStack(fluid, fluidQuantity))
}